components/ProjectModal/index.tsx

```diff
@@ -73,6 +85,85 @@ export default function ProjectModal({
           </div>
         </div>
         <div>{size}</div>
+        <AllKDMS>
+          {key_development_milestones?.map(
+            (milestone: Milestone, i: number) => {
+              let milestoneLabel = '';
+              let icon = <></>;
+              let kdmNumber = 0;
+
+              switch (milestone.milestoneTitle) {
+                case 'Entry to NYISO Queue':
+                  milestoneLabel = 'NYISO queue entered';
+                  kdmNumber = 1;
+                  break;
+                case 'Application for permit to ORES':
+                  milestoneLabel = 'ORES permit applied';
+                  kdmNumber = 2;
+                  break;
+                case 'Issuance of permit from ORES':
+                  milestoneLabel = 'ORES permit issued';
+                  kdmNumber = 3;
+                  break;
+                case 'Winning a contract award from NYSERDA':
+                  milestoneLabel = 'NYSERDA contracted';
+                  kdmNumber = 4;
+                  break;
+                case 'Execution of an Interconnection Agreement (IA)':
+                  milestoneLabel = 'IA executed';
+                  kdmNumber = 5;
+                  break;
+                case 'Start of operations':
+                  milestoneLabel = 'Operations begun';
+                  kdmNumber = 6;
+              }
+
+              if (milestone.completed) {
+                icon = (
+                  <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    width="8"
+                    height="7"
+                    viewBox="0 0 8 7"
+                    fill="none"
+                  >
+                    <path
+                      d="M7.66153 7H0.769226V6H7.66153V7ZM3.23076 4.71L1.26153 2.71L1.95569 2.005L3.23076 3.295L6.47507 0L7.16923 0.71L3.23076 4.71Z"
+                      fill="#ABBFF0"
+                    />
+                  </svg>
+                );
+              } else {
+                icon = (
```

Comment:
For code cleanliness sake, can we move some of this javascript code outside of the inline JSX. There's a lot to comprehend here if you don't have full context on what's going on.